src/server/pfs/server/api_server.go

```diff
@@ -451,7 +451,8 @@ func (a *apiServer) putFileObj(ctx context.Context, objClient obj.Client, reques
 		})
 		return eg.Wait()
 	}
-	return put(request.File.Path, url.Path)
+	// Joining Host and Path to retrieve the full path after "scheme://"
+	return put(request.File.Path, path.Join(url.Host, url.Path))
```

Comment:
Is this correct? It's been a while since I touched this code buy I seem to remember that the bucket gets passed directly to the `objClient` so I think by `Join`ing `url.Host` we're adding the bucket name to the front of the object we're requesting which means we won't find the object.